### ISSUE TITLE
Connect only working from one device

### ISSUE THREAD

**Issue by a13xmay** [michaelherger/Spotty-Plugin#58]
Hi Michael,

Not sure what's going on but my phone connects to Spotty via Connect fine but a couple of other devices do not work. The device shows it's connected however LMS doesn't play or show anything.

I have uploaded a log for you on your dropbox "alex server connect logging.log"

You help would be much appreciated.

**Comment by michaelherger** [comment]
This would happen in a few cases:

* other phones/devices aren't on the same network (mobile vs. wlan)
* "local announcements" are disabled or can't work due to some conflict on the host system (what OS are you running LMS on? What other services)?
* LMS is run in Docker (don't know the condition by heart)

Anything that might apply to your situation?

**Comment by a13xmay** [comment]
All devices def on the same wifi network. 
Not sure what are local announcements are?
Rubbing on a pretty vanilla version of Debian 9

**Comment by michaelherger** [comment]
Check the Spotty settings. there's a checkbox to "disable local announcements" (which can get checked automatically if it repeatedly fails to use them). When these announcements are disabled, only the account connected to LMS would be able to see the player.

**Comment by a13xmay** [comment]
No, this is not selected and there are no other options selected. I should re-iterate all devices can "see" the player. This isn't the issue.

I have just re-installed my whole system using Debian 11 64bit and LMS 8.3 and I'm experiencing the same.

**Comment by a13xmay** [comment]
Any thoughts Michael?

**Comment by KlausMu** [comment]
I have the same issue. No matter if the option
> Don't announce Squeezebox players running in Spotify Connect mode in your network. Check this option if you don't want your Spotify Connect enabled Squeezebox player to show up in all Spotify apps in your network.  

is set or not, I have the following behaviour:
I have two spotifiy accounts registered in "Spotty Spotify for Squeezebox". When I select "My Apps - Spotty - 'Account name' - Albums", the device **immediately** disappears in the Spotify Connect device list in one account and appears **at the same second** in the other account (checked with the Spotify Android app running on two phones with different accounts). No way to have the spotty device in two accounts at the same time.

Any hint?

**Comment by michaelherger** [comment]
Make sure you do NOT disallow announcements using that option. And once you've reset that switch, give it some time. And make sure your devices are all on the same network. Mobile phones sometimes stick to cell instead of switching to wifi etc.

**Comment by KlausMu** [comment]
I think I found the issue. I'm running LMS in docker. As mentioned [here](https://forums.slimdevices.com/showthread.php?114238-Spotty-4-4-8-not-showing-up-as-a-device-in-Spotify-app-when-running-LMS-in-Docker), Spotty only works correctly when it is running in host mode.
Only when running in host mode, I can see Spotty listening on these ports (sudo netstat -tulpn | grep LISTEN). Otherwise these entries are missing:
`tcp        0      0 0.0.0.0:35963           0.0.0.0:*               LISTEN      2196328/spotty`
`tcp        0      0 0.0.0.0:35397           0.0.0.0:*               LISTEN      2196330/spotty`   
`tcp        0      0 0.0.0.0:37369           0.0.0.0:*               LISTEN      2196327/spotty`

**Comment by sophof** [comment]
> I think I found the issue. I'm running LMS in docker. As mentioned [here](https://forums.slimdevices.com/showthread.php?114238-Spotty-4-4-8-not-showing-up-as-a-device-in-Spotify-app-when-running-LMS-in-Docker), Spotty only works correctly when it is running in host mode. Only when running in host mode, I can see Spotty listening on these ports (sudo netstat -tulpn | grep LISTEN). Otherwise these entries are missing: `tcp 0 0 0.0.0.0:35963 0.0.0.0:* LISTEN 2196328/spotty` `tcp 0 0 0.0.0.0:35397 0.0.0.0:* LISTEN 2196330/spotty` `tcp 0 0 0.0.0.0:37369 0.0.0.0:* LISTEN 2196327/spotty`

Thanks, this was the issue for me as well!

**Comment by michaelherger** [comment]
With the recent changes Spotify applied to their infrastructure I'll have to re-implement Connect mode from scratch. I therefore decided to remove that feature from the current release. Let's collect new reports once I have found a way to re-implement Connect.

Thanks for your understanding.
